File: api/BeSureApi/Models/Role.cs
using System.ComponentModel.DataAnnotations;

namespace BeSureApi.Models
{
    public class Role
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool IsActive { get; set; }
        public bool IsSystemRole { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
    }
    public class RoleList
    {
        public int Id { get; set; }
        public string RoleName { get; set; }
        public Boolean IsActive { get; set; }
    }
    public class RoleCreate
    {
        [Required(ErrorMessage = "validation_error_role_create_name_required")]
        [StringLength(64, ErrorMessage = "validation_error_role_create_name_max")]
        public string Name { get; set; }
        [Required(ErrorMessage = "validation_error_role_create_code_required")]
        [StringLength(16, ErrorMessage = "validation_error_role_create_code_max")]
        public string Code { get; set; }
        public string IsActive { get; set; }
    }
    public class UpdatedRoleDetail
    {
        public int RoleId { get; set; }
        public string Name { get; set; }
        public bool IsActive { get; set; }
    }
}

File: ui/src/types/role.ts

import { Decoder, array, boolean, nullable, number, object, string } from "decoders";

export interface RoleCreate {
    Name: string;
    Code:string;
    IsActive: string;
}

export interface RoleCreateResponse {
    IsRoleCreated: Boolean;
}

export const createDecoder: Decoder<RoleCreateResponse> = object({
    IsRoleCreated: boolean,
});

export interface RoleDetails {
    Id: number;
    Name: string;
    IsActive: boolean;
    IsSystemRole:boolean;
    CreatedBy: string;
    CreatedOn: string;
    UpdatedBy: null | string;
    UpdatedOn: null | string;
}

export const roleDetailsDecoder: Decoder<RoleDetails> = object({
    Id: number,
    Name: string,
    IsActive: boolean,
    IsSystemRole:boolean,
    CreatedBy: string,
    CreatedOn: string,
    UpdatedBy: nullable(string),
    UpdatedOn: nullable(string)
});
export interface RoleList {
    Roles: RoleDetails[],
    TotalRows: number;
    PerPage:number;
}

export const roleListDecoder: Decoder<RoleList> = object({
    Roles: array(roleDetailsDecoder),
    TotalRows: number,
    PerPage:number
});

export interface RoleTitleDetail {
    Id: number;
    RoleName: string;
}

export const roleDetailDecoder: Decoder<RoleTitleDetail> = object({
    Id: number,
    RoleName: string,
});

export interface RoleNames {
    RoleTitles: RoleTitleDetail[];
}

export const roleNamesDecoder: Decoder<RoleNames> = object({
    RoleTitles: array(roleDetailDecoder),
});

// Role Edit
export interface RoleEdit {
    RoleId: null | number
    Name: string;
    IsActive: boolean;
}

export interface RoleEditResponse {
    IsRoleUpdated: Boolean;
}

export const editDecoder: Decoder<RoleEditResponse> = object({
    IsRoleUpdated: boolean,
});
// Role Edit End

export interface RoleDeleteResponse {
    IsDeleted: boolean;
  }
  
  export const roleDeleteResponseDecoder: Decoder<RoleDeleteResponse> = object({
    IsDeleted: boolean,
  });